tests/unit/amo/components/TestCollectionManager.js

```diff
@@ -668,8 +668,27 @@ describe(__filename, () => {
     sinon.assert.calledWith(dispatchSpy, finishEditingCollectionDetails());
   });
 
-  it('calls history.goBack() on cancel when creating', () => {
-    const root = render({ creating: true });
+  it('calls history.push() when creating and language is defined', () => {
+    const siteLang = 'de';
+    const clientApp = 'firefox';
+    const localStore = dispatchClientMetadata({ clientApp, lang: siteLang })
+      .store;
+
+    const root = render({ creating: true, store: localStore });
+
+    simulateCancel(root);
+
+    sinon.assert.calledWith(
+      fakeHistory.push,
+      `/${siteLang}/${clientApp}/collections/`,
+    );
+  });
+
+  it('calls history.goBack() when creating and language is falsey', () => {
+    const siteLang = null;
```

Comment:
To match the Flow description, this should actually be `undefined`, not `null`, because `null` would be considered invalid by Flow.